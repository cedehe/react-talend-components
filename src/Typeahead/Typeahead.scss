$tc-typeahead-input-color: #000 !default;
$tc-typeahead-input-width: 262px;

$tc-typeahead-opened-items-container-color: #AAA !default;
$tc-typeahead-opened-items-container-background-color: #FFF !default;

$tc-typeahead-focused-item-background-color: #F7F7F7 !default;
$tc-typeahead-item-description-color: #C0C0C0 !default;

$tc-typeahead-section-border-color: #E0E0E0 !default;

$tc-typeahead-icon-color: #CCC !default;
$tc-typeahead-only-icon-color: #CCC !default;
$tc-typeahead-section-header-color: #63AEBD !default;

$tc-typeahead-items-list-shadow-color: #CCC;

/******************************************* container *************************************/
.tc-typeahead-container {
	position: relative;
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	width: auto;
}

.right-container {
	display: flex;
	flex-direction: column;
	align-items: flex-end;
}

/****************************************** input & icon ******************************************/
.typeahead-input {
	width: $tc-typeahead-input-width;
}

.typeahead-input-icon {
	position: relative;
	display: flex;
	align-items: center;

	.icon-cls {
		margin-left: -20px;
		color: $tc-typeahead-icon-color;
	}
}

.only-icon-cls {
	color: $tc-typeahead-only-icon-color;
}

/************************************* Container Open **************************************/
.items-container {
	display: none;
}

.container-open {
	.items-container {
		display: block;
		position: absolute;
		top: 100%;
		min-width: $tc-typeahead-input-width;
		background-color: $tc-typeahead-opened-items-container-background-color;
		border-radius: 4px;
		box-shadow: 1px 1px 5px 1px $tc-typeahead-items-list-shadow-color;
		max-height: 70vh;
		overflow-y: auto;
	}
}

/***************************************** item ********************************************/
.items-list {
	margin: 0;
	padding: 0;
	list-style-type: none;
}

.item {
	cursor: pointer;
	margin-bottom: 0;
	padding: 14px;

	.item-title {
		font-weight: normal;
	}

	.item-description {
		color: $tc-typeahead-item-description-color;
		font-size: 1.3rem;
		margin-bottom: 0;
	}
}

.item-focused {
	background-color: $tc-typeahead-focused-item-background-color;
}

/******************************************** section **************************************/
.section-header {
	align-items: center;
	color: $tc-typeahead-section-header-color;
	border-top: 1px solid $tc-typeahead-section-border-color;
	border-bottom: 1px solid $tc-typeahead-section-border-color;
	margin-top: -1px;
	padding: 14px;
	display: flex;

	.section-header-title {
		margin-left: 7px;
		font-size: 1.2rem;
		letter-spacing: 0.6px;
		text-transform: uppercase;
	}
}

/***************************************** highlight ***************************************/
.highlight-match {
	font-weight: 600;
	font-style: normal;
}
