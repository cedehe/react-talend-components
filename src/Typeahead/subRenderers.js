import React from 'react';
import { FormControl } from 'react-bootstrap';
import Icon from '../Icon';
import theme from './Typeahead.scss';

export const renderItem = (item, { value }) => {
	const splittedTitle = !value ? [item.title] : item.title.split(value);
	const emphasisedTitle = splittedTitle.map((title, index) => (
		<span key={index}>
			{title}
			{index !== splittedTitle.length - 1 ?
				<em className={theme['highlight-match']}>{value}</em> :
				''
			}
		</span>
	));

	return (
		<div className={theme.item}>
			<span className={theme['item-title']}>{emphasisedTitle}</span>
			<p className={theme['item-description']}>{item.description}</p>
		</div>
	);
};

export const renderInputComponent = config => (
	(props) => {
		let iconEl;
		if (config && config.icon) {
			iconEl = (
				<div className={theme['icon-cls']}>
					<Icon name={config.icon.name} title={config.icon.title} />
				</div>
			);
		}

		return (
			<div className={theme['typeahead-input-icon']}>
				<FormControl {...props} />
				{iconEl}
			</div>
		);
	}
);

export const renderSectionTitle = section => (
	<div className={theme['section-header']}>
		<Icon name={section.icon.name} title={section.icon.title} />
		<span className={theme['section-header-title']}>{section.title}</span>
	</div>
);
