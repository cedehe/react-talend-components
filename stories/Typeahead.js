import Autowhatever from 'react-autowhatever';
import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { FormControl, Nav } from 'react-bootstrap';
import { Typeahead } from '../src';

const exampleId = 'component-id';
const focusedItemIndex = 1;
const focusedSectionIndex = 0;
const value = 'le';
const renderItemData = { value };

const items = [
	{
		title: 'category 1',
		icon: {
			name: 'fa fa-filter',
			title: 'icon'
		},
		suggestions: [
			{
				title: 'le title 1',
				description: 'description: Uxoresque est in pacto est marito est hastam nomine in eos discessura incredibile tempus ardore.',
			},
			{
				title: 'title 2 les elephants elementaires ont des aile ',
				description: 'description: Aut aut cum satis inter Epicuri quidem cum erat inquam controversia autem mihi utrumque Attico.',
			},
		]
	},

	{
		title: 'category 2',
		icon: {
			name: 'fa fa-asterisk',
			title: 'icon'
		},
		suggestions: [
			{
				title: 'title 3',
				description: 'description: In sanciatur libere audeamus exspectemus amicitia et dum ne audeamus causa monendum honesta studium valeat.',
			},
		]
	},

	{
		title: 'category 3',
		icon: {
			name: 'fa fa-asterisk',
			title: 'icon'
		},
		suggestions: [
			{
				title: 'title 4',
				description: 'description: Praesentibus genero ne in Africani mandavi saepius ipsam C in libro et hoc Laeli cum.',
			},
			{
				title: 'title 5',
				description: 'description: Feceris unde tot illo tot clientes dederis numerando et indiscretus cum paria et unde ubi.',
			},
			{
				title: 'title 6',
				description: 'description: Gradu quos cedentium sunt appeterent ita ancoralia instar luna sunt etiam ubi incendente nihil observabant.',
			},
		]
	}];

let inputProps = {
	value,
	placeholder: 'Search anything',
};

// Input actions
const inputActionsList = ['onBlur', 'onKeyDown']; // TODO it is nice to have onChange event when the value is already given
inputProps = inputActionsList.reduce(
	(accumulator, funcName) =>
		Object.assign(accumulator, { [funcName]: action(`${funcName} event on input is triggered`) })
	, inputProps
);

// Item actions
const itemActionsList = ['onMouseEnter', 'onMouseLeave', 'onClick', 'onKeyDown'];
const itemProps = itemActionsList.reduce(
	(accumulator, funcName) =>
		Object.assign(accumulator, { [funcName]: action(`${funcName} event on item in the list is triggered`) })
	, {}
);

const props = {
	id: exampleId,
	config: {
		isOnlyIcon: true,
		icon: {
			name: 'fa fa-search',
			title: 'icon',
			actionStyle: 'link',
		},
		onInputIconClick: action('icon clicked'),
	},
	items: [],
};

storiesOf('Typeahead', module)
	.addWithInfo('Only Icon', () => {
		return (
			<div>
				<h2>Below is an example of a Typeahead</h2>
				<Typeahead {...props} />
			</div>
		);
	})

	.addWithInfo('Typeahead With Icon (LEFT ==> RIGHT)', () => {
		const myProps = Object.assign({}, props);
		myProps.config = {
			icon: {
				name: 'fa fa-search',
				title: 'icon',
			},
		};
		myProps.items = items;
		myProps.inputProps = inputProps;
		myProps.itemProps = itemProps;
		myProps.focusedItemIndex = focusedItemIndex;
		myProps.focusedSectionIndex = focusedSectionIndex;
		myProps.renderItemData = renderItemData;
		return (
			<div>
				<h2>Below is an example of a Typeahead</h2>
				<Typeahead {...myProps} />
			</div>
		);
	})

	.addWithInfo('Typeahead With Icon (RIGHT ==> LEFT)', () => {
		const myProps = Object.assign({}, props);
		myProps.config = {
			isOnTheRight: true,
			icon: {
				name: 'fa fa-search',
				title: 'icon',
			}
		};
		myProps.items = items;
		myProps.inputProps = inputProps;
		myProps.itemProps = itemProps;
		myProps.focusedItemIndex = focusedItemIndex;
		myProps.focusedSectionIndex = focusedSectionIndex;
		myProps.renderItemData = renderItemData;
		return (
			<div>
				<h2>Below is an example of a Typeahead</h2>
				<Typeahead {...myProps} />
			</div>
		);
	})

	.addWithInfo('Typeahead Without Icon (LEFT ==> RIGHT)', () => {
		const myProps = Object.assign({}, props);
		myProps.config = {};
		myProps.items = items;
		myProps.inputProps = inputProps;
		myProps.itemProps = itemProps;
		myProps.focusedItemIndex = focusedItemIndex;
		myProps.focusedSectionIndex = focusedSectionIndex;
		myProps.renderItemData = renderItemData;
		return (
			<div>
				<h2>Below is an example of a Typeahead</h2>
				<Typeahead {...myProps} />
			</div>
		);
	})

	.addWithInfo('Typeahead Without Icon (RIGHT ==> LEFT)', () => {
		const myProps = Object.assign({}, props);
		myProps.config = { isOnTheRight: true };
		myProps.items = items;
		myProps.inputProps = inputProps;
		myProps.itemProps = itemProps;
		myProps.focusedItemIndex = focusedItemIndex;
		myProps.focusedSectionIndex = focusedSectionIndex;
		myProps.renderItemData = renderItemData;
		return (
			<div>
				<h2>Below is an example of a Typeahead</h2>
				<Typeahead {...myProps} />
			</div>
		);
	})

	.addWithInfo('Typeahead With Icon & no items', () => {
		const myProps = Object.assign({}, props);
		myProps.config = {
			icon: {
				name: 'fa fa-search',
				title: 'icon',
			}
		};
		myProps.inputProps = inputProps;
		return (
			<div>
				<h2>Below is an example of a Typeahead</h2>
				<Typeahead {...myProps} />
			</div>
		);
	})
;
