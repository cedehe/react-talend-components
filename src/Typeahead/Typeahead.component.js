import React, { PropTypes } from 'react';
import classNames from 'classnames';
import Autowhatever from 'react-autowhatever';
import theme from './Typeahead.scss';
import { renderItem, renderInputComponent, renderSectionTitle } from './subRenderers';
import { Action } from '../Actions';

/**
 * @example
 <Typeahead
 id={exampleId}
 items={items}
 inputProps={inputProps}
 itemProps={itemProps}
 config={config}
 focusedItemIndex={focusedItemIndex}
 focusedSectionIndex={focusedSectionIndex}
 renderItemData={renderItemData}
 />
 */

function Typeahead({ config, ...eprops }) {
	if (config && config.isOnlyIcon) {
		return (
			<Action
				onClick={config.onInputIconClick}
				label={config.icon.title}
				hideLabel
				icon={config.icon.name}
				bsStyle={config.icon.actionStyle}
				className={theme['only-icon-cls']}
			/>);
	}

	const aweProps = { ...eprops };
	aweProps.items = aweProps.items ? aweProps.items : [];
	const containerClass = classNames(
		theme['tc-typeahead-container'],
		{
			[`${theme['right-container']}`]: aweProps.items.length && config.isOnTheRight,
		}
	);

	aweProps.renderItem = renderItem;
	aweProps.renderInputComponent = renderInputComponent(config);
	aweProps.renderSectionTitle = renderSectionTitle;
	aweProps.getSectionItems = section => section.suggestions;
	aweProps.multiSection = true;
	aweProps.theme = {
		container: containerClass,
		containerOpen: theme['container-open'],
		highlight: theme['highlight-match'],
		input: theme['typeahead-input'],
		itemFocused: theme['item-focused'],
		itemsContainer: theme['items-container'],
		itemsList: theme['items-list'],
		sectionContainer: theme['section-container'],
	};
	return (<Autowhatever {...aweProps} />);
}
Typeahead.propTypes = {
	config: PropTypes.shape({
		isOnlyIcon: PropTypes.boolean,
		icon: PropTypes.shape({
			name: PropTypes.string,
			title: PropTypes.string,
			actionStyle: PropTypes.string,
		}),
		onInputIconClick: PropTypes.func,
		isOnTheRight: PropTypes.boolean,
	}),
	focusedItemIndex: PropTypes.number,
	focusedSectionIndex: PropTypes.number,
	id: PropTypes.string,
	items: PropTypes.arrayOf(PropTypes.shape({
		title: PropTypes.string,
		description: PropTypes.string,
		suggestions: PropTypes.arrayOf(PropTypes.shape({
			title: PropTypes.string,
			description: PropTypes.string,
		})),
	})),
	inputProps: PropTypes.shape({
		value: PropTypes.string.isRequired,
		placeholder: PropTypes.string,
		onBlur: PropTypes.func,
		onKeyDown: PropTypes.func,
	}),
	itemProps: PropTypes.shape({
		onBlur: PropTypes.func,
		onKeyDown: PropTypes.func,
		onMouseEnter: PropTypes.func,
		onClick: PropTypes.func,
	}),
	renderItemData: PropTypes.shape({
		value: PropTypes.string,
	}),
};
export default Typeahead;
