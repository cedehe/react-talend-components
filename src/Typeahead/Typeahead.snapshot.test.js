import React from 'react';
import renderer from 'react-test-renderer';
import Typeahead from './Typeahead.component';

jest.mock('react-dom');

const exampleId = 'component-id';
const focusedItemIndex = 1;
const focusedSectionIndex = 0;
const value = 'le';
const renderItemData = { value };
const inputProps = {
	value,
	placeholder: 'Search anything',
};

const itemProps = {};

const items = [
	{
		title: 'category 1',
		icon: {
			name: 'fa fa-filter',
			title: 'icon',
		},
		suggestions: [
			{
				title: 'le title 1',
				description: 'description: Uxoresque est in pacto est marito est hastam nomine in eos discessura incredibile tempus ardore.',
			},
			{
				title: 'title 2 les elephants elementaires ont des aile ',
				description: 'description: Aut aut cum satis inter Epicuri quidem cum erat inquam controversia autem mihi utrumque Attico.',
			},
		],
	},

	{
		title: 'category 2',
		icon: {
			name: 'fa fa-asterisk',
			title: 'icon',
		},
		suggestions: [
			{
				title: 'title 3',
				description: 'description: In sanciatur libere audeamus exspectemus amicitia et dum ne audeamus causa monendum honesta studium valeat.',
			},
		],
	}];

const onlyIconProps = {
	id: exampleId,
	config: {
		isOnlyIcon: true,
		icon: {
			name: 'fa fa-search',
			title: 'icon',
		},
		onInputIconClick: jest.fn(),
	},
	items: [],
};

const typeaheadProps = {
	id: exampleId,
	config: {
		icon: {
			name: 'fa fa-search',
			title: 'icon',
			actionStyle: 'link',
		},
	},
	inputProps,
	itemProps,
	focusedItemIndex,
	focusedSectionIndex,
	renderItemData,
};


describe('Only icon', () => {
	it('should render only zoom icon', () => {
		// given
		const wrapper = renderer
			.create(<Typeahead {...onlyIconProps} />)
			.toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});
});

describe('Typeahead', () => {
	describe('icon', () => {
		it('should render typeahead with icon', () => {
			// when
			const wrapper = renderer
				.create(<Typeahead {...typeaheadProps} />)
				.toJSON();

			// then
			expect(wrapper).toMatchSnapshot();
		});

		it('should render typeahead without icon', () => {
			// given
			const myProps = Object.assign({}, typeaheadProps);
			myProps.config = {};

			// when
			const wrapper = renderer
				.create(<Typeahead {...myProps} />)
				.toJSON();

			// then
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('position', () => {
		it('should render typeahead (LEFT ==> RIGHT)', () => {
			// given
			const myProps = Object.assign({}, typeaheadProps);
			myProps.config = {};
			myProps.items = items;

			// when
			const wrapper = renderer
				.create(<Typeahead {...myProps} />)
				.toJSON();

			// then
			expect(wrapper).toMatchSnapshot();
		});
		// The case when the Typeahead is from RIGHT ==> LEFT is not covered by snapshots,
		// because currently it is not possible to import theme in snapshots.
	});

	describe('items', () => {
		it('should render typeahead with items', () => {
			// given
			const myProps = Object.assign({}, typeaheadProps);
			myProps.config = {};
			myProps.items = items;

			// when
			const wrapper = renderer
				.create(<Typeahead {...myProps} />)
				.toJSON();

			// then
			expect(wrapper).toMatchSnapshot();
		});

		it('should render typeahead without items', () => {
			// given
			const myProps = Object.assign({}, typeaheadProps);
			myProps.config = {};

			// when
			const wrapper = renderer
				.create(<Typeahead {...myProps} />)
				.toJSON();

			// then
			expect(wrapper).toMatchSnapshot();
		});

		it('should render typeahead items with match', () => {
			// given
			const myProps = Object.assign({}, typeaheadProps);
			myProps.config = {};
			myProps.items = items;
			myProps.renderItemData = renderItemData;

			// when
			const wrapper = renderer
				.create(<Typeahead {...myProps} />)
				.toJSON();

			// then
			expect(wrapper).toMatchSnapshot();
		});
	});
});
