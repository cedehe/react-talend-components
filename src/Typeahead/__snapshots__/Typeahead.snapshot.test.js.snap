exports[`Only icon should render only zoom icon 1`] = `
<button
  aria-describedby={42}
  className="btn btn-default"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  role={null}
  type="button">
  <i
    aria-hidden="true"
    className="fa fa-search"
    title={null} />
</button>
`;

exports[`Typeahead icon should render typeahead with icon 1`] = `
<div
  style={Object {}}>
  <div
    className={undefined}>
    <input
      aria-activedescendant="react-autowhatever-component-id-section-0-item-1"
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup={false}
      aria-owns="react-autowhatever-component-id"
      autoComplete="off"
      className="form-control"
      id={undefined}
      onKeyDown={undefined}
      placeholder="Search anything"
      role="combobox"
      style={Object {}}
      type="text"
      value="le" />
    <div
      className={undefined}>
      <i
        aria-hidden="true"
        className="fa fa-search"
        title="icon" />
    </div>
  </div>
  <div
    id="react-autowhatever-component-id"
    style={Object {}} />
</div>
`;

exports[`Typeahead icon should render typeahead without icon 1`] = `
<div
  style={Object {}}>
  <div
    className={undefined}>
    <input
      aria-activedescendant="react-autowhatever-component-id-section-0-item-1"
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup={false}
      aria-owns="react-autowhatever-component-id"
      autoComplete="off"
      className="form-control"
      id={undefined}
      onKeyDown={undefined}
      placeholder="Search anything"
      role="combobox"
      style={Object {}}
      type="text"
      value="le" />
  </div>
  <div
    id="react-autowhatever-component-id"
    style={Object {}} />
</div>
`;

exports[`Typeahead items should render typeahead items with match 1`] = `
<div
  style={Object {}}>
  <div
    className={undefined}>
    <input
      aria-activedescendant="react-autowhatever-component-id-section-0-item-1"
      aria-autocomplete="list"
      aria-expanded={true}
      aria-haspopup={true}
      aria-owns="react-autowhatever-component-id"
      autoComplete="off"
      className="form-control"
      id={undefined}
      onKeyDown={undefined}
      placeholder="Search anything"
      role="combobox"
      style={Object {}}
      type="text"
      value="le" />
  </div>
  <div
    id="react-autowhatever-component-id"
    style={Object {}}>
    <div
      style={Object {}}>
      <div
        style={Object {}}>
        <div
          className={undefined}>
          <i
            aria-hidden="true"
            className="fa fa-filter"
            title="icon" />
          <span
            className={undefined}>
            category 1
          </span>
        </div>
      </div>
      <ul
        role="listbox"
        style={Object {}}>
        <li
          id="react-autowhatever-component-id-section-0-item-0"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 1
                
              </span>
            </span>
            <p
              className={undefined}>
              description: Uxoresque est in pacto est marito est hastam nomine in eos discessura incredibile tempus ardore.
            </p>
          </div>
        </li>
        <li
          id="react-autowhatever-component-id-section-0-item-1"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 2 
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                s e
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                phants e
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                mentaires ont des ai
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 
                
              </span>
            </span>
            <p
              className={undefined}>
              description: Aut aut cum satis inter Epicuri quidem cum erat inquam controversia autem mihi utrumque Attico.
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      style={Object {}}>
      <div
        style={Object {}}>
        <div
          className={undefined}>
          <i
            aria-hidden="true"
            className="fa fa-asterisk"
            title="icon" />
          <span
            className={undefined}>
            category 2
          </span>
        </div>
      </div>
      <ul
        role="listbox"
        style={Object {}}>
        <li
          id="react-autowhatever-component-id-section-1-item-0"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 3
                
              </span>
            </span>
            <p
              className={undefined}>
              description: In sanciatur libere audeamus exspectemus amicitia et dum ne audeamus causa monendum honesta studium valeat.
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Typeahead items should render typeahead with items 1`] = `
<div
  style={Object {}}>
  <div
    className={undefined}>
    <input
      aria-activedescendant="react-autowhatever-component-id-section-0-item-1"
      aria-autocomplete="list"
      aria-expanded={true}
      aria-haspopup={true}
      aria-owns="react-autowhatever-component-id"
      autoComplete="off"
      className="form-control"
      id={undefined}
      onKeyDown={undefined}
      placeholder="Search anything"
      role="combobox"
      style={Object {}}
      type="text"
      value="le" />
  </div>
  <div
    id="react-autowhatever-component-id"
    style={Object {}}>
    <div
      style={Object {}}>
      <div
        style={Object {}}>
        <div
          className={undefined}>
          <i
            aria-hidden="true"
            className="fa fa-filter"
            title="icon" />
          <span
            className={undefined}>
            category 1
          </span>
        </div>
      </div>
      <ul
        role="listbox"
        style={Object {}}>
        <li
          id="react-autowhatever-component-id-section-0-item-0"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 1
                
              </span>
            </span>
            <p
              className={undefined}>
              description: Uxoresque est in pacto est marito est hastam nomine in eos discessura incredibile tempus ardore.
            </p>
          </div>
        </li>
        <li
          id="react-autowhatever-component-id-section-0-item-1"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 2 
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                s e
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                phants e
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                mentaires ont des ai
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 
                
              </span>
            </span>
            <p
              className={undefined}>
              description: Aut aut cum satis inter Epicuri quidem cum erat inquam controversia autem mihi utrumque Attico.
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      style={Object {}}>
      <div
        style={Object {}}>
        <div
          className={undefined}>
          <i
            aria-hidden="true"
            className="fa fa-asterisk"
            title="icon" />
          <span
            className={undefined}>
            category 2
          </span>
        </div>
      </div>
      <ul
        role="listbox"
        style={Object {}}>
        <li
          id="react-autowhatever-component-id-section-1-item-0"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 3
                
              </span>
            </span>
            <p
              className={undefined}>
              description: In sanciatur libere audeamus exspectemus amicitia et dum ne audeamus causa monendum honesta studium valeat.
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Typeahead items should render typeahead without items 1`] = `
<div
  style={Object {}}>
  <div
    className={undefined}>
    <input
      aria-activedescendant="react-autowhatever-component-id-section-0-item-1"
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup={false}
      aria-owns="react-autowhatever-component-id"
      autoComplete="off"
      className="form-control"
      id={undefined}
      onKeyDown={undefined}
      placeholder="Search anything"
      role="combobox"
      style={Object {}}
      type="text"
      value="le" />
  </div>
  <div
    id="react-autowhatever-component-id"
    style={Object {}} />
</div>
`;

exports[`Typeahead position should render typeahead (LEFT ==> RIGHT) 1`] = `
<div
  style={Object {}}>
  <div
    className={undefined}>
    <input
      aria-activedescendant="react-autowhatever-component-id-section-0-item-1"
      aria-autocomplete="list"
      aria-expanded={true}
      aria-haspopup={true}
      aria-owns="react-autowhatever-component-id"
      autoComplete="off"
      className="form-control"
      id={undefined}
      onKeyDown={undefined}
      placeholder="Search anything"
      role="combobox"
      style={Object {}}
      type="text"
      value="le" />
  </div>
  <div
    id="react-autowhatever-component-id"
    style={Object {}}>
    <div
      style={Object {}}>
      <div
        style={Object {}}>
        <div
          className={undefined}>
          <i
            aria-hidden="true"
            className="fa fa-filter"
            title="icon" />
          <span
            className={undefined}>
            category 1
          </span>
        </div>
      </div>
      <ul
        role="listbox"
        style={Object {}}>
        <li
          id="react-autowhatever-component-id-section-0-item-0"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 1
                
              </span>
            </span>
            <p
              className={undefined}>
              description: Uxoresque est in pacto est marito est hastam nomine in eos discessura incredibile tempus ardore.
            </p>
          </div>
        </li>
        <li
          id="react-autowhatever-component-id-section-0-item-1"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 2 
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                s e
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                phants e
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                mentaires ont des ai
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 
                
              </span>
            </span>
            <p
              className={undefined}>
              description: Aut aut cum satis inter Epicuri quidem cum erat inquam controversia autem mihi utrumque Attico.
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      style={Object {}}>
      <div
        style={Object {}}>
        <div
          className={undefined}>
          <i
            aria-hidden="true"
            className="fa fa-asterisk"
            title="icon" />
          <span
            className={undefined}>
            category 2
          </span>
        </div>
      </div>
      <ul
        role="listbox"
        style={Object {}}>
        <li
          id="react-autowhatever-component-id-section-1-item-0"
          role="option"
          style={Object {}}>
          <div
            className={undefined}>
            <span
              className={undefined}>
              <span>
                tit
                <em
                  className={undefined}>
                  le
                </em>
              </span>
              <span>
                 3
                
              </span>
            </span>
            <p
              className={undefined}>
              description: In sanciatur libere audeamus exspectemus amicitia et dum ne audeamus causa monendum honesta studium valeat.
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
